In [1]:
import numpy as np

In [2]:
(-1/2)*(np.log2(1/2))

0.5

In [3]:
(-1/2)*(np.log2(1/2))

0.5

In [4]:
0.5+0.5

1.0

Entropy of parent node is 1 by adding the values

# Entropy of X left child

In [5]:
(-1/1)* (np.log2(1/1))

-0.0

# Entropy of X Right child

In [6]:
(-2/3)*(np.log2(2/3))

0.38997500048077083

In [7]:
(-1/3)*(np.log2(1/3))

0.5283208335737187

In [8]:
0.38997500048077083 + 0.5283208335737187

0.9182958340544896

# Information Gain

In [9]:
0.9182958340544896 *3/4

0.6887218755408672

In [10]:
1/4* -0.0

-0.0

In [11]:
0.6887218755408672+ 0

0.6887218755408672

In [12]:
1-0.6887218755408672

0.31127812445913283

 the information gain for X child is 0.31127812445913283

# Child node y 

In [13]:
(-2/2) * (np.log2(2/2))

-0.0

In [14]:
+(-2/2) *(np.log2(2/2))

-0.0

In [15]:
0+0

0

entropy for y child node is 0

In [16]:
1-0

1

 The information gain is 1

# Child node Z

In [17]:
(-1/2)*(np.log2(1/2))

0.5

In [18]:
(-1/2)*(np.log2(1/2))

0.5

In [19]:
0.5+0.5

1.0

In [20]:
2/4*1

0.5

In [21]:
2/4*1

0.5

In [23]:
1-1

0